### PR TITLE
chm-406 undb action view page

### PR DESCRIPTION
**Pull request by liyuanmontreal**
### General description
migrate UNDB Action view page from chm project

### Designs
same design as chm project

### Testing instructions
search->UNDB Action

this is new page 
image

this is old chm project page

image

### CODE REVIEW

File: app/views/forms/view/chm/undb-action.vue [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    const props = defineProps({
+        documentInfo: { type: Object, required: true },
+        locale      : { type:String}
+    })
+
+    const { t } = useI18n({ messages });    
+   
+    const country = computed(()=>{
+        const term = ref({});
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/undb-action.vue:124]: why we need  const term = ref({}); inside computed ?

File: app/views/forms/view/chm/undb-action.vue [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                    <ng v-vue-ng:km-value-ml  :value="document.description" :locales="locale" html></ng>  
+                </div>
+
+                <div v-if="document.descriptionNative">
+                    <label for="subTopic">{{ t("shortDescriptionInOfficialLanguage") }}</label> 
+                    <ng v-vue-ng:km-value-ml  :value="document.descriptionNative" :locales="locale" html></ng>  
+                </div>
+
+                <div v-if="document.notes">
+                    <label for="subTopic">{{ t("commentsOrAdditionalInfo") }}</label>
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/undb-action.vue:90]: <label for="subTopic" ?

each label typically corresponds to a specific element, remove this for="subTopic" from all labels if not required.

File: app/views/forms/view/chm/undb-action.vue (hunk 1 above)
Comment (liyuanmontreal) [reply app/views/forms/view/chm/undb-action.vue]: change to outside

File: app/views/forms/view/chm/undb-action.vue (hunk 2 above)
Comment (liyuanmontreal) [reply app/views/forms/view/chm/undb-action.vue]: removed

**Comment by liyuanmontreal** [comment]
add links for email and url.
change img url  to get  image under thumb folder
image

File: app/views/forms/view/chm/undb-action.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    import { ref,computed } from 'vue'; 
+    import '~/components/scbd-angularjs-controls/form-control-directives/km-value-ml.js'
+    import kmTerm from '~/components/km/KmTerm.vue';
+    import messages from '~/app-text/views/reports/chm/undb-action.json';
+    import { useI18n } from 'vue-i18n';
+    import { formatDate } from '~/components/kb/filters';
+
+    const props = defineProps({
+        documentInfo: { type: Object, required: true },
+        locale      : { type:String}
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/undb-action.vue:129]: If we have 'locale' in useI18n(), then why do we need to pass it as a prop?
const { t, locale } = useI18n({ messages });

File: app/views/forms/view/chm/undb-action.vue [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                <div v-if="document.title">
+                    <label>{{ t("title") }}</label>
+                    <ng v-vue-ng:km-value-ml  :value="document.title" :locales="locale" html></ng>  
+                </div>
+            </section>
+
+            <section>                
+                <div v-if="document.startDate">
+                    <label>{{ t("startDate") }}</label>
+                    <ng v-vue-ng:km-value-ml  :value="formatDate(document.startDate, 'YYYY-MM-DD')" :locales="locale" html></ng>
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/undb-action.vue:14]: do we need ':locales="locale"' and  'html' for date ?

Comment (liyuanmontreal) [reply app/views/forms/view/chm/undb-action.vue]: fixed

**Review by tahmeedshah — APPROVED** [review]

**Review by tahmeedshah — APPROVED** [review]

**Review by tahmeedshah — APPROVED** [review]

File: app/views/forms/view/chm/undb-action.vue [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                    <label >{{ t("descriptionInOfficialLanguage") }}</label> 
+                    <div class="km-value km-pre">
+                        {{ document.descriptionNative }}
+                    </div>
+                </div>
+               
+                <div v-if="document.notes">
+                    <label >{{ t("comments") }}</label>                      
+                    <ng v-vue-ng:km-value-ml  :value="document.notes" :locales="locale" html km-pre></ng>  
+                </div>
```
Comment (blaisefonseca) [inline app/views/forms/view/chm/undb-action.vue:112]: notes should not be displayed for public, they only appear in edit form
```suggestion

```

**Pull request merged**